plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/decoders/ArrayDecoder.java

```diff
@@ -34,21 +37,46 @@ public ArrayDecoder(Decoder elementDecoder)
     }
 
     @Override
-    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
+    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output, List<String> dereferenceName)
     {
         Object data = getter.get();
 
         if (data == null) {
             output.appendNull();
         }
         else if (data instanceof List<?> list) {
-            ((ArrayBlockBuilder) output).buildEntry(elementBuilder -> list.forEach(element -> elementDecoder.decode(hit, () -> element, elementBuilder)));
+            ((ArrayBlockBuilder) output).buildEntry(elementBuilder -> list.forEach(element -> elementDecoder.decode(hit, () -> element, elementBuilder, dereferenceName)));
+        }
+        else if ((data instanceof Map nestedFields) && !dereferenceName.isEmpty()) {
+            String topDereferenceName = dereferenceName.getFirst();
+            decode(hit, () -> getField(nestedFields, topDereferenceName), output, dereferenceName.subList(1, dereferenceName.size()));
         }
         else {
-            ((ArrayBlockBuilder) output).buildEntry(elementBuilder -> elementDecoder.decode(hit, () -> data, elementBuilder));
+            ((ArrayBlockBuilder) output).buildEntry(elementBuilder -> elementDecoder.decode(hit, () -> data, elementBuilder, dereferenceName));
         }
     }
 
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null) {
+            return false;
+        }
+        if (!(o instanceof ArrayDecoder arrayDecoder)) {
+            return false;
+        }
+        return arrayDecoder.elementDecoder.equals(elementDecoder);
```

Comment:
I think It would be good to have the implementation of `equals` as other connector has. 

https://github.com/trinodb/trino/blob/d3bb9848c963a81216be250628f77ef54b9f7bbd/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnProjectionInfo.java#L96-L108
```suggestion
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        ArrayDecoder that = (ArrayDecoder) o;
        return Objects.equals(this.elementDecoder, that.elementDecoder);
```

Reply:
Same comment for other decoders's `equals` implementation.